### DISCUSSION TITLE
Execute Query (F5) - Run Selected DAX query if Selected, otherwise run All (Mimic DAX Studio & SSMS Behavior)

### DISCUSSION THREAD

**Discussion by corycundy** [TabularEditor/TabularEditor3#1349] [Ideas]
### Is your feature request related to a problem? Please describe

The feature is related to a development behavior many have acquired over time in which F5 (or some shortcut key) can be used to either run all of the queries in an open query window or the selected query.

DAX Studio has this functionality split between Execute and Execute Selection. There are so many times when I highlight a single query of many and press F5 and inadvertently run all of the queries, which can be undesirable.

### Describe the solution you'd like

I would prefer some setting or in some way to have the ability of the Execute (F5) to run the queries selected if there are any selected, otherwise run all the queries in the query window. This would mimic the behavior in DAX Studio and  SQL Server Management Studio in which I would guess many TE3 users have a history with.

I don't believe the current split of functionality between F5 and Shift F5 allows a way to configure the desired behavior

### Goals

1. Even if this isn't the default behavior, allowing a preference for Execute to do both would be great.

**Comment by MariaJoseFF** [comment]
Hi @corycundy,

Thanks for your valuable feedback. I understand your concern since I have been an SSMS user and pressed F5 many times on auto-pilot. Unfortunately, we currently can't set up F5 to execute the entire script and the selection on Tabular Editor, but I have made a note of this request.

We currently have a couple of ways to run and apply the selected DAX on DAX Queries and DAX Scripts:

- When using a DAX Query:
   1. You can press the "Execute selection" button.
![image](https://github.com/user-attachments/assets/e02e101d-d98d-4f6d-94a4-fb3cedd36ffa)
   2. You can use one of the [Shortcuts for DAX Queries](https://docs.tabulareditor.com/te3/features/shortcuts.html#dax-query), which executes the selection: Shift+F5.


- When using a DAX Script:
   1. You can press the "Apply selection" button.
![image](https://github.com/user-attachments/assets/dee3559a-ffbf-4226-a352-18fa0f432ef2)
   2. You can use one of the [Shortcuts for DAX Scripts](https://docs.tabulareditor.com/te3/features/shortcuts.html#dax-script), which applies the selection: F8.
   
Regarding the **F5 shortcut**. It is possible to [Customize Shortcuts](https://docs.tabulareditor.com/te3/features/shortcuts.html#customizing-shortcuts), but one shortcut can't be assigned to multiple commands. Nevertheless, a workaround you could try is to reassign the shortcut that executes the selection in a DAX Query to use "F5". The command for this is called "Editor.DaxQuery.ExecuteSelection". But then you will always need to have a selection if there is something you want to run. For example, if you want to run the whole script, you would need to press Ctrl+A and then F5 after reassigning the shortcut.

To make this reassignment, you will need to do the following:
![Animation](https://github.com/user-attachments/assets/17f9cb63-20c7-4192-b139-1536195ece4b)

I hope this helps!
